### PR TITLE
Add docs on GPU memory allocation.

### PR DESCRIPTION
**Pull request by skye**

### CODE REVIEW

File: docs/gpu_memory_allocation.rst [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+**Running multiple JAX processes concurrently.**
+  Either use XLA_PYTHON_CLIENT_MEM_FRACTION to give each process an appropriate
+  amount of memory, or set XLA_PYTHON_CLIENT_PREALLOCATE=false.
+
+**Running JAX and GPU TensorFlow concurrently.**
+  TensorFlow also preallocates by default, so this is similar to running
+  multiple JAX processes concurrently. One solution is to use CPU-only
+  TensorFlow (e.g. if you're only doing data loading with TF). Alternatively,
+  use XLA_PYTHON_CLIENT_MEM_FRACTION or XLA_PYTHON_CLIENT_PREALLOCATE.
```
Comment (hawkinsp) [inline docs/gpu_memory_allocation.rst]: You might want to point to the equivalent TensorFlow option.

File: docs/gpu_memory_allocation.rst [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+**Running JAX on the display GPU.**
+  Use XLA_PYTHON_CLIENT_MEM_FRACTION or XLA_PYTHON_CLIENT_PREALLOCATE.
```
Comment (hawkinsp) [inline docs/gpu_memory_allocation.rst]: Not directly related to this PR, but I still suspect we need to do something to the defaults here; JAX on GPU reliably crashes on first operation on my workstation.

**Review by hawkinsp — APPROVED** [review]
Looks great, thanks for doing this!

File: docs/gpu_memory_allocation.rst (hunk 1 above)
Comment (skye) [reply docs/gpu_memory_allocation.rst]: Done.

Comment (skye) [reply docs/gpu_memory_allocation.rst]: Oops I forgot to actually commit that addition, gonna push that change directly to master.

**Pull request merged**